packages/local_auth/local_auth_darwin/lib/local_auth_darwin.dart

```diff
@@ -40,7 +42,9 @@ class LocalAuthDarwin extends LocalAuthPlatform {
             biometricOnly: options.biometricOnly,
             sticky: options.stickyAuth,
             useErrorDialogs: options.useErrorDialogs),
-        _pigeonStringsFromAuthMessages(localizedReason, authMessages));
+        defaultTargetPlatform == TargetPlatform.macOS
```

Comment:
This should be using `Platform.is*` (or a local-test-overridable wrapper around that), since it's interfacing with native code; if non-test code is executing this code with the default target overridden, it still needs to use the strings that correspond to the actual native code that will execute.